src/Compilers/CSharp/Portable/CodeGen/EmitAddress.cs

```diff
@@ -546,7 +546,7 @@ private LocalDefinition EmitInstanceFieldAddress(BoundFieldAccess fieldAccess, A
             //      taking field addresses, so we have to turn Constrained into writeable.
             var tempOpt = EmitReceiverRef(fieldAccess.ReceiverOpt, addressKind == AddressKind.Constrained ? AddressKind.Writeable : addressKind);
 
-            _builder.EmitOpCode(ILOpCode.Ldflda);
+            _builder.EmitOpCode(field.RefKind == RefKind.None ? ILOpCode.Ldflda : ILOpCode.Ldfld);
```

Comment:
>ILOpCode.Ldfld [](http://example.com/codeflow?start=82&length=14)

The name of the function feels somewhat confusing now

Reply:
I agree the name is somewhat confusing, but it seems similar to `EmitLocalAddress()` which simply loads the local if the local is a `ref`.